gef.py

```diff
@@ -10737,8 +10737,22 @@ def read_cstring(self,
         try:
             res_bytes = self.read(address, length)
         except gdb.error:
-            err(f"Can't read memory at '{address}'")
-            return ""
+            current_address = address
+            res_bytes = b""
+            while len(res_bytes) < length:
+                try:
+                    new_end_addr = current_address + DEFAULT_PAGE_SIZE
```

Comment:
I don't think that this variable name is very good, since it's not where we end at all, it's just a page away, and we are about to chop it down with the mask